### ISSUE TITLE
Okular does not opens (flatpak)

### ISSUE THREAD

**Issue by aquaspy** [flathub/org.kde.okular#84]
Well, not sure if there is some incompatibility in void linux, but there is something making it impossible to use okular flatpak here.

Every time I try to open it I get an error:
![image](https://user-images.githubusercontent.com/49214136/86128188-6be25d00-bab7-11ea-98a6-8df60bab6287.png)

Any ideas about this? Thanks!

**Comment by tsdgeos** [comment]
Interesting, can you please give us details of your system?

i.e. which distribution, etc.

**Comment by Erick555** [comment]
I see **void linux** in description ([this one](https://voidlinux.org/))

**Comment by tsdgeos** [comment]
Ah right, missed that. 

Can you please past the contents of flatpak list ?

**Comment by aquaspy** [comment]
> Ah right, missed that.
> 
> Can you please past the contents of flatpak list ?

Of course!
![image](https://user-images.githubusercontent.com/49214136/86267558-43cd2980-bb9d-11ea-9d0a-318f4a0405c9.png)

I think KDE dependencies are ok

**Comment by tsdgeos** [comment]
Looks reasonable.

Can you do this?
flatpak run --command=sh --devel org.kde.okular 
gdb -ex run /app/bin/okular-bin 

it will hopefully crash, then type bt and paste the output of bt

**Comment by aquaspy** [comment]
> Looks reasonable.
> 
> Can you do this?
> flatpak run --command=sh --devel org.kde.okular
> gdb -ex run /app/bin/okular-bin
> 
> it will hopefully crash, then type bt and paste the output of bt

![image](https://user-images.githubusercontent.com/49214136/86306334-6b96ae80-bbea-11ea-93bf-2b698cbba86e.png)

I won't even crash I guess. I think I need to install kde sdk?

**Comment by aquaspy** [comment]
![image](https://user-images.githubusercontent.com/49214136/86306727-6423d500-bbeb-11ea-8164-c67c62ce60aa.png)
Ok Im installing

**Comment by aquaspy** [comment]
Sadly, even after the install, the error still the same
![image](https://user-images.githubusercontent.com/49214136/86306824-92a1b000-bbeb-11ea-93aa-f22f0e315d60.png)

**Comment by Erick555** [comment]
Could you show `flatpak run -vvv org.kde .okular`?

**Comment by aquaspy** [comment]
![image](https://user-images.githubusercontent.com/49214136/86469137-33d55700-bd0f-11ea-8e51-83f44842f8ec.png)

I use glibc version of void, so no musl fault I guess.

**Comment by Erick555** [comment]
Do you have `pulseaudio` and `xdg-desktop-portal-{kde/gtk}` installed?

**Comment by aquaspy** [comment]
> Do you have `pulseaudio` and `xdg-desktop-portal-{kde/gtk}` installed? https://docs.voidlinux.org/config/external-applications.html#flatpak

Yes, xdg-desktop-portal for both gtk and kde are installed. Pulseaudio is too. I still not sure on what is causing this.

Does this flatpak depends on opnssl?

**Comment by Erick555** [comment]
You shouldn't need both gtk and kde portal unless you use multiple DE interchangeably. Flatpak apps have own openssl and everything else they need however I'm not sure if the crash doesn't happen in flatpak itself before real app was started.

**Comment by tsdgeos** [comment]
Which flatpak version are you running?

**Comment by aquaspy** [comment]
Flatpak 1.6.3

> Which flatpak version are you running?

**Comment by aquaspy** [comment]
> You shouldn't need both gtk and kde portal unless you use multiple DE interchangeably. Flatpak apps have own openssl and everything else they need however I'm not sure if the crash doesn't happen in flatpak itself before real app was started.

oh, got it. Sadly, I don't know what is the issue then.

**Comment by tsdgeos** [comment]
Ok, invoking @alexlarsson now.

Any idea why if would crash when doing  `flatpak run --command=sh --devel org.kde.okular`

Seems like it's flatpak itself crashing? See https://user-images.githubusercontent.com/49214136/86306824-92a1b000-bbeb-11ea-93aa-f22f0e315d60.png

**Comment by aquaspy** [comment]
@tsdgeos  I am not on void linux anymore, just switched to artix with runit. I installed okular from here to test and seems to work perfectly. Maybe a void issue?

**Comment by tsdgeos** [comment]
Yes, Void was always the suspect either because they do something wrong or because their weirdness makes flatpak crash

**Comment by AmamosV** [comment]
I'm having a similar problem. I'm on a chromebook, using the linux on chromeos. I installed okular with flatpak and it doesn't launch. When I click on the icon it starts like if it was going to load, but then never finishes, not erros given. I uninstalled and installed with apt-get and then it worked perfectly.

**Comment by AmamosV** [comment]
Update, reinstalled okular with flatpak and tried to use the run command instead of the chromebook's icon to see what it would turn out. That's what i got:

> $ flatpak run org.kde.okular
> qt.qpa.xcb: could not connect to display 
> qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
> This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
> 
> Available platform plugins are: eglfs, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, xcb.

I also use several other apps on linux in my chromebook and they are all working fine with flatpak, just getting this on okular.

Thanks

**Comment by Justinzobel** [comment]
Is this issue still occurring for you?

**Comment by Justinzobel** [comment]
As there has been no response, I am closing this issue. If you are still experiencing it, please leave a comment here.
